streams-tests/shared/src/test/scala/zio/stream/ZSinkSpec.scala

```diff
@@ -242,11 +293,29 @@ object ZSinkSpec extends ZIOBaseSpec {
             assertZIO((stream ++ stream).rechunk(3).run(sink))(equalTo(List(1, 2, 3, 4)))
           }
         ),
-        test("head")(
-          check(Gen.listOf(Gen.small(Gen.chunkOfN(_)(Gen.int)))) { chunks =>
-            val headOpt = ZStream.fromChunks(chunks: _*).run(ZSink.head[Int])
-            assertZIO(headOpt)(equalTo(chunks.flatMap(_.toSeq).headOption))
-          }
+        suite("head")(
+          test("basic")(
+            check(Gen.listOf(Gen.small(Gen.chunkOfN(_)(Gen.int)))) { chunks =>
+              val headOpt = ZStream.fromChunks(chunks: _*).run(ZSink.head[Int])
+              assertZIO(headOpt)(equalTo(chunks.flatMap(_.toSeq).headOption))
+            }
+          ),
+          test("late error")(
+            assertZIO {
+              (ZStream(1, 2, 3) ++ ZStream.fail("Aie") ++ ZStream(5, 1, 2, 3, 4, 5))
```

Comment:
@kyri-petrou , here you can see the same 'error handling' behavior with the `head` sink in case of a 'late' failure